File: 프로그래머스/lv2/42584. 주식가격/주식가격.py
def solution(prices):
    stack = []
    answer = [0] * len(prices)
    
    for i, v in enumerate(prices):
        while stack and prices[stack[-1]] > v:
            tok = stack.pop()
            answer[tok] = i - tok
        stack.append(i)
    
    for s in stack:
        answer[s] = len(prices) - 1 - s
    
    return answer

File: 프로그래머스/Lv.2/42584. 주식가격/주식가격.java
import java.util.Stack;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Stack<Integer> stack = new Stack<>();
        
        for (int index = 0; index < prices.length; index++) {
            while (!stack.isEmpty() && prices[stack.peek()] > prices[index]) {
                int target = stack.pop();
                answer[target] = index - target;
            }
            
            stack.add(index);
        }
        
        for (int n : stack) {
            answer[n] = (prices.length - 1) - n;
        }
        
        return answer;
    }
}